contracts/NodeVersionBeacon.cdc

```diff
@@ -116,94 +120,133 @@ pub contract ExecutionNodeVersionBeacon {
+                NodeVersionBeacon.versionTable.keys.contains(blockHeight): "No boundary defined at that blockHeight."
+                NodeVersionBeacon.upcomingBlockBoundaries.contains(blockHeight): "Boundary not defined in upcomingBlockBoundaries."
             }
 
             // Ensure deletion occurs with enough time for nodes to respond
             assert(
-                blockHeight > getCurrentBlock().height + ExecutionNodeVersionBeacon.versionUpdateBuffer,
+                blockHeight > getCurrentBlock().height + NodeVersionBeacon.versionUpdateBuffer,
                 message: "Cannot delete version boundary within update buffer period or historical mappings."
             )
 
             // Remove the version mapping and upcomingBlockBoundaries then emit event
-            let removed: Semver = ExecutionNodeVersionBeacon.versionTable.remove(
+            let removed: Semver = NodeVersionBeacon.versionTable.remove(
                 key: blockHeight
             )!
-            ExecutionNodeVersionBeacon.upcomingBlockBoundaries.remove(
-                at: ExecutionNodeVersionBeacon.upcomingBlockBoundaries.firstIndex(of: blockHeight)!
+            NodeVersionBeacon.upcomingBlockBoundaries.remove(
+                at: NodeVersionBeacon.upcomingBlockBoundaries.firstIndex(of: blockHeight)!
             )
 
-            emit ExecutionNodeVersionTableDeletion(height: blockHeight, version: removed)
-
+            //emit NodeVersionTableDeletion(height: blockHeight, version: removed)
             // Clean up upcoming & archived block boundaries before
             // returning removed version for verification
-            ExecutionNodeVersionBeacon.archiveOldBlockBoundaries()
+            NodeVersionBeacon.archiveOldBlockBoundaries()
             return removed
         }
 
+        /// Separate method to emit the NodeVersionTableUpdated event
+        ///
+        /// @param newVersions
+        /// @param deletedVersions
+        ///
+        pub fun emitNodeVersionTableUpdated() {
+            
+            let updatedTable: {UInt64: Semver} = {}
+
+            // Assure that there is at least one working version for the current block
+            let currentVersionBlock = NodeVersionBeacon.getNextVersionBoundaryPair()[0] as! UInt64
+
+            updatedTable[currentVersionBlock] = NodeVersionBeacon.versionTable[currentVersionBlock]
+
+            for versionBlock in NodeVersionBeacon.versionTable.keys {
+                if (versionBlock > currentVersionBlock) {
+                    updatedTable[versionBlock] = NodeVersionBeacon.versionTable[versionBlock]
+                }
+            }
+
+            emit NodeVersionTableUpdated(versionTable: updatedTable,
+                sequence: NodeVersionBeacon.nextTableUpdatedSequence)
+
+            NodeVersionBeacon.nextTableUpdatedSequence = NodeVersionBeacon.nextTableUpdatedSequence + 1
+
+        }
```

Comment:
Can you add some more comments here to describe what this is doing?

Reply:
Absolutely, sorry about that, I didn't considered the code done until Maks have check on it (I actually have to introduce some more changes to the function)